src/Solution/ExplicitSolution.f90

```diff
@@ -0,0 +1,394 @@
+  !!
+  !<
+  subroutine sln_da(this)
+    ! -- modules
+    use MemoryManagerModule, only: mem_deallocate
+    ! -- dummy variables
+    class(ExplicitSolutionType) :: this !< ExplicitSolutionType instance
+    !
+    ! -- lists
+    call this%modellist%Clear()
+    deallocate (this%modellist)
+    !
+    !
+    ! -- Scalars
+    call mem_deallocate(this%id)
+    call mem_deallocate(this%iu)
+    call mem_deallocate(this%ttsoln)
+    call mem_deallocate(this%icnvg)
+    !
+    ! -- return
+    return
+  end subroutine sln_da
+
+  !> @ brief Solve solution
+  !!
+  !!  Solve the models in this solution for kper and kstp.
+  !!
+  !<
+  subroutine sln_ca(this, isgcnvg, isuppress_output)
+    ! -- modules
+    ! -- dummy variables
+    class(ExplicitSolutionType) :: this !< ExplicitSolutionType instance
+    integer(I4B), intent(inout) :: isgcnvg !< solution group convergence flag
+    integer(I4B), intent(in) :: isuppress_output !< flag for suppressing output
+    ! -- local variables
+    class(BaseModelType), pointer :: mp => null()
+    character(len=LINELENGTH) :: line
+    character(len=LINELENGTH) :: fmt
+    integer(I4B) :: im
+
+    ! advance the models and solution
+    call this%prepareSolve()
+
+    select case (isim_mode)
+    case (MVALIDATE)
+      line = 'mode="validation" -- Skipping assembly and solution.'
+      fmt = "(/,1x,a,/)"
+      do im = 1, this%modellist%Count()
+        mp => GetBaseModelFromList(this%modellist, im)
+        call mp%model_message(line, fmt=fmt)
+      end do
+    case (MNORMAL)
+
+      ! solve the models
+      call this%solve()
+
+      ! finish up
+      call this%finalizeSolve(isgcnvg, isuppress_output)
+    end select
+    !
+    ! -- return
+    return
+
+  end subroutine sln_ca
+
+  !> @ brief prepare to solve
+  !!
+  !!  Prepare for the system solve by advancing the simulation.
+  !!
+  !<
+  subroutine prepareSolve(this)
+    ! -- dummy variables
+    class(ExplicitSolutionType) :: this !< ExplicitSolutionType instance
+    ! -- local variables
+    integer(I4B) :: im
+    class(BaseModelType), pointer :: mp => null()
+
+    ! -- Model advance
+    do im = 1, this%modellist%Count()
+      mp => GetBaseModelFromList(this%modellist, im)
+      call mp%model_ad()
+    end do
+
+    ! advance solution
+    call this%sln_ad()
+
+  end subroutine prepareSolve
+
+  !> @ brief Build and solve the simulation
+  !!
+  !! Builds and solves the system for this explicit solution.
+  !! It roughly consists of the following steps
+  !! (1) ! kluge note: list steps
+  !!
+  !<
+  subroutine solve(this)
+    ! -- dummy variables
+    class(ExplicitSolutionType) :: this !< ExplicitSolutionType instance
+    ! -- local variables
+    class(BaseModelType), pointer :: mp => null()
+    integer(I4B) :: im
+    real(DP) :: ttsoln
+    !
+    ! -- particle solve
+    call code_timer(0, ttsoln, this%ttsoln)
+    do im = 1, this%modellist%Count()
+      mp => GetBaseModelFromList(this%modellist, im)
```

Comment:
I think that we should consider if ExplicitSolution has a list of ExplicitModel types, just NumericalSolution holds a list of NumericalModel types. However, no need to stall this PR for that.